conctr.agent.lib.nut

```diff
@@ -195,18 +228,393 @@ class Conctr {
+            topics = ["dev/" + _device_id];
+        }
+
+        if (typeof topics != "array") {
+            topics = [topics];
+        }
+
+        local action = "subscribe";
+        local headers = {};
+        local payload = {};
+        local chunks = "";
+        local contentLength = null;
+        local reqTime = time();
+        local reconnect = function() {
+                subscribe(topics, cb);
+            }
+
+
+        // Http streaming callback
+        local _streamCb = function(chunk) {
+
+            // User called unsubscribe. Close connection.
+            if (_pollingReq == null) return;
+
+            // accumulate chuncks till we get a full msg
+            chunks += chunk;
+
+            // Check whether we have received the content length yet (sent as first line of msg)
+            if (contentLength == null) {
+                // Sweet, we want to extract it out, itll be
+                // chilling just before the /r/n lets find it
+                local eos = chunks.find(STREAM_TERMINATOR);
+                // Got it! 
+                if (eos != null) {
+                    // Pull it out
+                    contentLength = chunks.slice(0, eos);
+                    try {
+                        contentLength = contentLength.tointeger();
+                    } catch (e) {
+                        // server.error("Conctr: " + e);
+                    }
+
+                    // Leave the rest of the msg
+                    chunks = chunks.slice(eos + STREAM_TERMINATOR.len());
+                }
+            }
+
+            // Handle incorrect message i.e. 502 html returned
+            if (typeof contentLength == "string") {
+                // server.error("Conctr: Got invalid response: " + chunks)
+                return reconnect();
+            }
+
+            if (contentLength != null && chunks.len() >= contentLength) {
+                // Got a full msg, process it!
+                _processData(chunks.slice(0, contentLength), cb);
+
+                // Get any partial chunks if any and keep waiting for the end of new message
+                chunks = chunks.slice(contentLength + STREAM_TERMINATOR.len());
+                contentLength = null;
+            }
+        }
+
+        // http done callback
+        local _doneCb = function(resp) {
+            
+            // We dont allow non chunked requests. So if we recieve a message in this func
+            // it is the last message of the steam and may contain the last chunk
+            if ("body" in resp && !(resp.body == null || resp.body == "")) {
+                _streamCb(resp);
+            }
+
+            local wakeupTime = 0;
+
+            if (resp.statuscode >= 200 && resp.statuscode <= 300) {
+                // wake up time is 0
+            } else if (resp.statuscode == 429) {
+                wakeupTime = 1;
+            } else if(resp.statuscode == 599){
+                server.log("Blowing zombie agents brains out!");
```

Comment:
Really?

599 just means "MQTT server disconnected". You should retry with a wakeup time of 0.
It doesn't mean "You got disconnected by another client with the same clientid".